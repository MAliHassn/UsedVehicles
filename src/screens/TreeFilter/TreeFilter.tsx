import { Pressable, StyleSheet, Text, View } from 'react-native';
import React from 'react';
import { Colors } from '../../constants/Colors';
import { Fonts } from '../../constants/Fonts';
import Header from '../../components/Header';
import Filter from '../../components/Filter';
import LinearGradient from 'react-native-linear-gradient';
import { useFilter } from './Hooks';

const TreeFilter = () => {
  const { realTimeChanges,
    clickOnNodes,
    filter_data,
    checkedNode,
    checkedChilds,
    returnRow
  } = useFilter()

  return (
    <View style={styles.container}>

      <Header />

      <View style={styles.filter}>
        <Filter
          key={"test"}
          data={filter_data as Node[]}
          onPress={(v) => { realTimeChanges(v) }}
          onPressNode={(v) => { clickOnNodes(v) }}
        />
      </View>

      <LinearGradient
        colors={["rgba(0,0,0,0.3)", "rgba(0,0,0,0.2)",]}
        style={styles.selected}
        start={{ x: 1, y: 0 }}
        end={{ x: 0.2, y: 0.9 }}
      >

        {returnRow(
          "tree ids:",
          checkedChilds?.ids,
          checkedNode?.id
        )}
        {returnRow(
          "tree titles:",
          checkedChilds?.titles,
          checkedNode?.title
        )}

      </LinearGradient>

      <LinearGradient
        colors={[Colors.secondary, Colors.primary,]}
        style={styles.apply}
        start={{ x: 1, y: 0 }}
        end={{ x: 0.2, y: 0.9 }}
      >
        <Pressable onPress={() => { console.log("") }}>
          <Text style={styles.text}>{"APPLY"}</Text>
        </Pressable>
      </LinearGradient>

    </View>
  );
};

export default TreeFilter;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.white
  },
  filter: {
    padding: 10,
    marginBottom: 110,
    paddingTop: 0,
    flex: 1
  },
  selected: {
    justifyContent: "center",
    alignItems: "center",
    position: "absolute",
    bottom: 70,
    elevation: 8,
    left: 0,
    right: 0,
    height: 40,
    marginHorizontal: 10,
  },
  apply: {
    justifyContent: "center",
    alignItems: "center",
    position: "absolute",
    bottom: 10,
    elevation: 8,
    left: 0,
    right: 0,
    height: 50,
    marginHorizontal: 10
  },
  row: {
    flexDirection: "row",
    justifyContent: "center"
  },
  text: {
    color: Colors.white,
    fontSize: Fonts.heading
  },
  subtext: {
    color: Colors.black,
    fontSize: Fonts.sub_body
  },

});