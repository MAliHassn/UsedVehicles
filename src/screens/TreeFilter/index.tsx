export { default as TreeFilter } from './TreeFilter';