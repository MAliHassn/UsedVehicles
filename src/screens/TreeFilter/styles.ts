import {StyleSheet} from 'react-native';
import {useCustomTheme} from '../../constants/useCustomTheme';

const useStyles = () => {
  const {colors, fonts} = useCustomTheme();
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: colors.white,
    },
    filter: {
      padding: 10,
      marginBottom: 110,
      paddingTop: 0,
      flex: 1,
    },
    selected: {
      justifyContent: 'center',
      alignItems: 'center',
      position: 'absolute',
      bottom: 80,
      elevation: 8,
      left: 0,
      right: 0,
      height: 40,
      marginHorizontal: 10,
    },
    apply: {
      justifyContent: 'center',
      alignItems: 'center',
      position: 'absolute',
      bottom: 20,
      elevation: 8,
      left: 0,
      right: 0,
      height: 50,
      marginHorizontal: 10,
    },
    row: {
      flexDirection: 'row',
      justifyContent: 'center',
    },
    text: {
      color: colors.white,
      fontSize: fonts.heading,
    },
    subtext: {
      color: colors.black,
      fontSize: fonts.sub_body,
    },
  });

  return {
    styles,
  };
};

export default useStyles;
