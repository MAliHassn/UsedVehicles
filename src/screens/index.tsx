export { Splash } from './Splash';
export { VehiclesList } from './VehiclesList';
export { TreeFilter } from './TreeFilter';
