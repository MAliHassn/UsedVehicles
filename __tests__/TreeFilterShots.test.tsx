import React from 'react';
import renderer from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';
import {TreeFilter} from '../src/screens/TreeFilter';
import { waitFor } from '@testing-library/react-native';

it('renders correctly', async() => {
  const component = (
    <NavigationContainer>
      <TreeFilter />
    </NavigationContainer>
  );
  const rendered = renderer.create(component).toJSON();
  await waitFor(() => expect(rendered).toMatchSnapshot());
});