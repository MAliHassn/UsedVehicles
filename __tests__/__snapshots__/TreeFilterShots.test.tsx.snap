// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginTop": -10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "height": 50,
          "justifyContent": "center",
          "width": 50,
        }
      }
    >
      <Image
        resizeMethod="auto"
        source={
          {
            "testUri": "../../../src/assets/ui/back.png",
          }
        }
        style={
          {
            "aspectRatio": 1,
            "height": 22,
            "width": 22,
          }
        }
      />
    </View>
    <Image
      resizeMethod="auto"
      source={
        {
          "testUri": "../../../src/assets/ui/applogo.png",
        }
      }
      style={
        {
          "aspectRatio": 2,
          "height": 70,
          "width": "auto",
        }
      }
    />
  </View>
  <View
    style={
      {
        "flex": 1,
        "marginBottom": 110,
        "padding": 10,
        "paddingTop": 0,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "paddingBottom": 20,
        }
      }
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingVertical": 5,
              }
            }
          >
            <Image
              resizeMethod="auto"
              source={
                {
                  "testUri": "../../../src/assets/ui/add.png",
                }
              }
              style={
                {
                  "aspectRatio": 1,
                  "height": 22,
                  "width": 22,
                }
              }
            />
            <Text
              style={
                {
                  "color": "#666666",
                  "fontSize": 12,
                  "paddingStart": 3,
                }
              }
            >
              Cars
            </Text>
          </View>
        </View>
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingVertical": 5,
              }
            }
          >
            <Image
              resizeMethod="auto"
              source={
                {
                  "testUri": "../../../src/assets/ui/add.png",
                }
              }
              style={
                {
                  "aspectRatio": 1,
                  "height": 22,
                  "width": 22,
                }
              }
            />
            <Text
              style={
                {
                  "color": "#666666",
                  "fontSize": 12,
                  "paddingStart": 3,
                }
              }
            >
              Trucks
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <BVLinearGradient
    colors={
      [
        1291845632,
        855638016,
      ]
    }
    endPoint={
      {
        "x": 0.2,
        "y": 0.9,
      }
    }
    locations={null}
    startPoint={
      {
        "x": 1,
        "y": 0,
      }
    }
    style={
      {
        "alignItems": "center",
        "bottom": 70,
        "elevation": 8,
        "height": 40,
        "justifyContent": "center",
        "left": 0,
        "marginHorizontal": 10,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 10,
          }
        }
      >
        tree ids:
      </Text>
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 10,
          }
        }
      >
        -&gt;
         
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 10,
          }
        }
      >
        tree titles:
      </Text>
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 10,
          }
        }
      >
        -&gt;
         
      </Text>
    </View>
  </BVLinearGradient>
  <BVLinearGradient
    colors={
      [
        4291559424,
        4288217088,
      ]
    }
    endPoint={
      {
        "x": 0.2,
        "y": 0.9,
      }
    }
    locations={null}
    startPoint={
      {
        "x": 1,
        "y": 0,
      }
    }
    style={
      {
        "alignItems": "center",
        "bottom": 10,
        "elevation": 8,
        "height": 50,
        "justifyContent": "center",
        "left": 0,
        "marginHorizontal": 10,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          {
            "color": "#ffffff",
            "fontSize": 18,
          }
        }
      >
        APPLY
      </Text>
    </View>
  </BVLinearGradient>
</View>
`;
